ompi/mca/sharedfp/sm/sharedfp_sm_file_open.c

```diff
@@ -135,7 +135,6 @@ int mca_sharedfp_sm_file_open (struct ompi_communicator_t *comm,
         return OMPI_ERROR;
     }
 
-        free(sm_filename);
```

Comment:
:smile: It's probably fine to include it on this PR, but it's probably a good idea to go ahead and make it a separate commit (ping me offline if you wanna see the git magic to split this out into a separate commit).  Thanks!